### DISCUSSION TITLE
Bookmarks

### DISCUSSION THREAD

**Discussion by stephenwf** [digirati-co-uk/madoc-platform#465] [Ideas]
Bookmarks are, in their simplest form, a way for normal users of Madoc to save things they may be interested in to view later. However, with the foundation of Madoc this could end up being a very powerful feature for casual users and power-users alike.

### Basic bookmarking
<div align="center">
<img width="434" alt="Screenshot 2021-10-24 at 11 53 26" src="https://user-images.githubusercontent.com/8266711/138590792-a538eb46-4681-42a6-9754-a1d7a6aff097.png">
</div>

A simple approach would be a button on images or manifests, allowing you to bookmark the page you're on. You could then, from your profile, see these bookmarks.

### Bookmark collections
The next step would be to allow users to maintain different lists, group together relevant pieces of content together.

<div align="center"><img width="414" alt="Screenshot 2021-10-24 at 11 58 19" src="https://user-images.githubusercontent.com/8266711/138590927-3dfde87a-0591-44d3-b9a1-a5e144d050f7.png"></div>

Offering a simple way to create these when creating a bookmark.
 
<div align="center">
<img width="411" alt="Screenshot 2021-10-24 at 11 59 19" src="https://user-images.githubusercontent.com/8266711/138590966-330f46f0-2735-46e1-ac83-f9bbd68aa721.png">
</div>

Already this would offer some utility to users who want to gather items for research or just to come back to. This could be used in a crowdsourcing setting to plan work. 

### Viewing bookmarks
Where bookmarks can really build on the foundation of Madoc is the repackaging of bookmark collections into various rich viewing experiences. With experiences like simple pages listing the content and thumbnails or exporting to IIIF and embedding existing tools from the community.

* **Mirador workspaces** - You go to your bookmark collection, you choose "+ new mirador workspace". You see your bookmarks down one side, and you can drag a few into a multi-up view. As pull new images into the workspaces, zoom and pan around - your changes are saved. The next time you go to your bookmarks, you can jump back to your workspace.
* **Manifest editor** - You go to your bookmark collection, you choose "+ create manifest". You will see the manifest editor, and in the content panel - all of your bookmarks. You can pull in canvases from your bookmarks and compose a new manifest.
* **Markdown composer** - utilising the IIIF image API, compose markdown documents containing cropped and zoomed images from your bookmarks, alongside your own text. 
* **View in [ ... ]** - Open your collection, or a manifest from your collection in a 3rd-party viewer. 

### Sharing bookmarks

With the "apps" above, there's an opportunity to be creative or productive and share the outputs. Sharing a mirador workspace, or blog post draft in markdown, or maybe just a simple list of images you've collected.

**Comment by NLW-paulm** [comment]
From the start we have been a big advocate for bookmarks, particularly as filtering and searching with the existing platform was difficult. When users were working their way through a large collection, being able to bookmark an item to get back to it quickly was a really important step.  However a lot of the reasons for needing bookmarks have been replaced by better functionality in 2.0.

Having said that, what is proposed above opens up a whole new usage for the bookmarks, particularly for people wanting to write or research across our collections.  The suggestions would actually (potentially) allow us to use the Madoc platform for more than just crowdsourcing in its purest form.  For example, research students could request access to certain content we have in IIIF and then be presented with it through madoc for them to use/research/collect.

We don't really have any systems that use Markdown publicly, but being able to create content/blog post/web page from a saved collection in some form could be really handy from researchers and a Marketing point of view. We probably need to look at options for this, Markdown, rich text, basic HTML?

Being able to share the output is also a good call, even if it is just a manifest URL.  We need to think of ways of 'saving' that manifest so it's perpetually available especially if we are talking research needs. It probably needs to live somewhere outside of madoc.

There is also the opportunity of using this system for researchers to create citations from collections.

**Further Development Ideas -**
I wonder if this bookmark feature (or saved items to align it with social tools??) could actually be rolled out into its own API?  To allow us to integrate it into other viewers and projects. For example if someone searching our newspapers (all IIIF enabled) wanted to save something it would save it into their account in madoc/Torf.  And then allow to reuse and combine with our interface.  As said above I think this feature could take this beyond traditional crowdsourcing and allow people to make more use of our collections across the board.
